pkg/controller/clone/common.go

```diff
@@ -0,0 +1,277 @@
+package clone
+
+import (
+	"context"
+	"fmt"
+	"sort"
+
+	snapshotv1 "github.com/kubernetes-csi/external-snapshotter/client/v6/apis/volumesnapshot/v1"
+	corev1 "k8s.io/api/core/v1"
+	storagev1 "k8s.io/api/storage/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/client-go/tools/record"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	cdiv1 "kubevirt.io/containerized-data-importer-api/pkg/apis/core/v1beta1"
+	"kubevirt.io/containerized-data-importer/pkg/common"
+	cc "kubevirt.io/containerized-data-importer/pkg/controller/common"
+	"kubevirt.io/containerized-data-importer/pkg/util"
+)
+
+// IsDataSourcePVC checks for PersistentVolumeClaim source kind
+func IsDataSourcePVC(kind string) bool {
+	return kind == "PersistentVolumeClaim"
+}
+
+// IsDataSourceVolumeClone checks for VolumeCloneSourceRef source kind
+func IsDataSourceVolumeClone(dataSource *corev1.TypedObjectReference) bool {
+	return dataSource != nil && dataSource.Kind == cdiv1.VolumeCloneSourceRef
+}
+
+// GetVolumeCloneSource returns the VolumeCloneSource dataSourceRef
+func GetVolumeCloneSource(ctx context.Context, c client.Client, pvc *corev1.PersistentVolumeClaim) (*cdiv1.VolumeCloneSource, error) {
+	if !IsDataSourceVolumeClone(pvc.Spec.DataSourceRef) {
+		return nil, fmt.Errorf("invalid dataSourceRef")
+	}
+
+	ns := pvc.Namespace
+	if pvc.Spec.DataSourceRef.Namespace != nil {
+		ns = *pvc.Spec.DataSourceRef.Namespace
+	}
+
+	obj := &cdiv1.VolumeCloneSource{}
+	exists, err := getResource(ctx, c, ns, pvc.Spec.DataSourceRef.Name, obj)
+	if err != nil {
+		return nil, err
+	}
+
+	if !exists {
+		return nil, nil
+	}
+
+	return obj, nil
+}
+
+// AddCommonLabels adds common labels to a resource
+func AddCommonLabels(obj metav1.Object) {
+	if obj.GetLabels() == nil {
+		obj.SetLabels(make(map[string]string))
+	}
+	obj.GetLabels()[common.CDILabelKey] = common.CDILabelValue
+}
+
+// AddCommonClaimLabels adds common labels to a pvc
+func AddCommonClaimLabels(pvc *corev1.PersistentVolumeClaim) {
+	AddCommonLabels(pvc)
+	if util.ResolveVolumeMode(pvc.Spec.VolumeMode) == corev1.PersistentVolumeFilesystem {
+		pvc.GetLabels()[common.KubePersistentVolumeFillingUpSuppressLabelKey] = common.KubePersistentVolumeFillingUpSuppressLabelValue
+	}
+}
+
+// AddOwnershipLabel adds owner label
+func AddOwnershipLabel(label string, obj, owner metav1.Object) {
+	if obj.GetLabels() == nil {
+		obj.SetLabels(make(map[string]string))
+	}
+	obj.GetLabels()[label] = string(owner.GetUID())
+}
+
+// IsSourceClaimReady checks that PVC exists, is bound, and is not being used
+func IsSourceClaimReady(ctx context.Context, c client.Client, namespace, name string) (bool, error) {
```

Comment:
Another minor complaint, but since we consistently use `PVC` or `PersistentVolumeClaim` in our code, I think it can be inconsistent to use claim in all these new functions. I think it wouldn't hurt to change `Claim` to `PVC`, wdyt?

Reply:
I prefer to be consistent across the whole repo, but if Kubernetes is using claim in its code then I'm ok if we start using it in new code.